### ISSUE TITLE
License incompatibility with Inkscape

### ISSUE THREAD

**Issue by jazzynico** [oberstet/scour#7]
Hi Tobias,

I just found out that you're the new scour maintainer, and it's great you took over a project that was inactive for years.
One of the last changes applied to the old branch fixed a license compatibility issue with Inkscape. The question has been discussed in the Inkscape board list (https://www.mail-archive.com/inkscape-board@lists.sourceforge.net/msg00033.html) and the fix applied revision 222 (http://bazaar.launchpad.net/~scouring/scour/trunk/revision/222#scour.py).
Would you be willing to dual-license your branch too, so that we can use it in Inkscape?
Thanks in advance!

**Comment by oberstet** [comment]
In general, I don't have a problem with that. Could you expand why the dual-licensing is necessary for Inkscape to use scour? E.g. should Inkscape invoke scour as a command line tool, it is my understanding og the GPL, that this wouldn't constitute a "derived work" (and hence wouldn't require dual-licensing).

If we do it, we need prior written consent of all contributors to do so.

That is all contributors here https://github.com/oberstet/scour/graphs/contributors *and* all contributors to scour in history.

**Comment by jazzynico** [comment]
Inkscape is still considered a GPLv2 project, and thus, if I understand correctly (quoting https://www.mail-archive.com/inkscape-board@lists.sourceforge.net/msg00014.html):
"The creator of the code published under Apache License 2.0 can prohibit a publishing of his code under GPL 2.0. So we are not allowed to publish it at all."

Note that the original author already agreed to dual-license its branch.

Also note that I'm not a license expert. If needed, I can ask the Inkscape board for details.

**Comment by oberstet** [comment]
As said, if a GPL program invokes an external program (starts a new OS process by invoking the respective command), the license of the external program is irrelevant. Running the external program in this way does *not* make the combination into a derived work.

We need permission from *all* contributors, not only the original author.

**Comment by Ede123** [comment]
@jazzynico and @oberstet: Could you please figure this out? I just wanted to start hacking on Scour regarding some issues that were bugging me for a long time. I'm hesitant to start though if it is not guaranteed that the code can be used in Inkscape.

Also relevant in this context: issue #11 to answer the question if this actually is the official scour project or just a fork.

**Comment by oberstet** [comment]
@Ede123 @jazzynico It is my understanding that

a) Inkscape invokes scour by forking scour as an external program, providing command line options and possible consuming stdout. But that's it. In particular scour isn't "linked" into Inkscape or what.
b) Because of a), the license of the program invoked (scour) is irrelevant. The GPL can't "infect" a program forked and talked to over command line options / stdin/stdout.

And because of a) and b), the whole issue of dual-licensing scour or changing the scour license is void. So I'd close this issue and call it a day.

Now, _if_ a) would be wrong or you don't agree to b): this would require license change to scour. And to do that in a waterproof way, we have to ask each and every contributor! Otherwise it's legally bogus. But again, I say: it's void, close the issue, everything is fine.

**Comment by oberstet** [comment]
@Ede123 as a sidenote to #11, I should repeat: this repo is the source of the scour package published to PyPI - and I did ask Jeff in advance if that is ok. He was fine. So if a repo that publishes to PyPI is the "official" repo, then it is this repo.

**Comment by Ede123** [comment]
Since the question just came up again I'm going to write down my personal opinion for future reference and I tend to agree with @oberstet:
* The relicensing that was done in the Launchpad repository probably was not legitimate and therefore not a real solution to start with. If anybody has any evidence that points in the other direction please state so. I assume if we can get a relicensing done properly @oberstet would agree?
* To the best of my knowledge we should be fine to use Scour in Inkscape anyway!

The second bullet point is supported by the following material:
* As long as we understand Inkscape and Scour as two different programs (and I believe this a completely valid assumption) we can assume the inclusion of Scour to be a "mere aggregate", see GPLv2 FAQ on [Mere Aggregation](http://www.gnu.org/licenses/old-licenses/gpl-2.0-faq.en.html#MereAggregation). Therefore the license of one has no effect on the other program.
* Even if we would consider Scour as a "plugin" for Inkscape (which is hardly the case - Scour is an independent module) this could at worst be considered a borderline case (see GPLv2 FAQ on [GPL and Plugins](http://www.gnu.org/licenses/old-licenses/gpl-2.0-faq.en.html#GPLAndPlugins)).

In conclusion: While a proper relicensing surely would be the preferable option and resolve any doubts, we should be perfectly fine to use Scour as-is.
